epyqlib/devicetreeview.py

```diff
@@ -231,6 +242,70 @@ def add_device(self, bus, device=None):
 
         return device
 
+    def check_software_hash(self, node):
+        pcan_bus = can.interface.Bus(bustype=node.interface, channel=node.channel)
+
+        # Recieve N messages on the PCAN bus.
+        # Discover the node ID from the arbitration ID.
+        # Store the found node ID's in a set.
+        node_id_set = set()
+        for i in range(100):
+            recv_id = pcan_bus.recv(0.1)
+            if recv_id is not None:
+                id = recv_id.arbitration_id & 0xFF
+                node_id_set.add(id)
+
```

Comment:
Some portion of what is below is going to need to be repeated for each node id so as to get hash for each of them, I think?  I don't recall off hand if the device's node-id can reasonably be changed on the fly presently.  Even if it can, the hash getting needs repeated.

Or, perhaps that's a second PR and I'm thinking ahead.

Reply:
Again, maybe this is my misunderstanding of how this should all work. I'm hoping this is a good enough starting point and we can iron out all these details in the next PR.

Reply:
As I read it presently this 1) collects a list of node IDs actively talking on the bus and 2) gets the software hash for the default node ID (247).  If 247 isn't present, I think this will fail out with an exception dialog.  If that's the goal in the first pass then it looks good to me.  These are certainly pieces of the (much) bigger goal.

A related side note, I noticed that `epyqlib.hildevice.Device` doesn't seem to offer any way to specify the node ID.

Reply:
Yes, you are right. I see your point. I don't know enough to say whether `epyqlib.hildevice.Device` will continue to be used to retrieve the software hash, so I don't know if it makes sense to update that code to specify the node ID? 

Another question is if the check for node ID finds more than one node ID, what to do?

Reply:
I was thinking the software hash would be acquired for each found node.  But, I'm fine breaking this up into whatever pieces.